web/src/copy_and_paste.ts

```diff
@@ -265,7 +285,7 @@ export function analyze_selection(selection) {
     };
 }
 
-function get_end_tr_from_endc($endc) {
+function get_end_tr_from_endc($endc: JQuery<Node>): JQuery {
     if ($endc.attr("id") === "bottom_whitespace" || $endc.attr("id") === "compose_close") {
         // If the selection ends in the bottom whitespace, we should
         // act as though the selection ends on the final message.
```

Comment:
The logic here looks buggy as well; `$(".message_row").last();` does not filter to the currently visible message list, if the combined feed is still present/invisible. 

@N-Shar-ma perhaps you can do a follow-up PR fixing this and the other apparent logic bug this TS migration reveals? I don't think there's any regressions in this PR.

Reply:
Fixed in #30096